hyperspy/decorators.py

```diff
@@ -106,3 +110,123 @@ def wrapper(self, *args, **kwargs):
         else:
             cm(self, *args, **kwargs)
     return wrapper
+
+
+class deprecated:
+    """Decorator to mark deprecated functions with an informative
+    warning.
+    Adapted from
+    `scikit-image
+    <https://github.com/scikit-image/scikit-image/blob/main/skimage/_shared/utils.py>`_
+    and `matplotlib
+    <https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/_api/deprecation.py>`_.
```

Comment:
When copying code from somewhere else, we should make sure to comply with the license of the original code...

Reply:
Do you mean the license of scikit-image and Matplotlib, or kikuchipy? I wrote this part in kikuchipy, and I give @CSSFrancis permission to include this piece of code and for any contributor to HyperSpy to modify it without including the license of kikuchipy in HyperSpy.

The [deprecation tools in kikuchipy](https://github.com/pyxem/kikuchipy/blob/develop/kikuchipy/_util/_deprecated.py) were not copied from [scikit-image](https://github.com/scikit-image/scikit-image/blob/e37e48d5dc137fceb548d28efdbe2d9a4d4dd316/skimage/_shared/utils.py#L357) or [Matplotlib](https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/_api/deprecation.py), rather, I looked to their decorators for inspiration. I realize that the word "adapted" is less descriptive and something like "inspired" is better.

Reply:
I think that we should be good then?  I also think that the scikit-image and Matplotlib licenses are very permissive so we shouldn't have any issues there...